pykafka/balancedconsumer.py

```diff
@@ -437,6 +443,21 @@ def _rebalance(self):
                     # If retrying, be sure to make sure the
                     # partition allocation is correct.
                     participants = self._get_participants()
+                    if self._consumer_id not in participants:
+                        self._consumer.stop()
+                        # zookeeper failure, all ephemeral nodes for this consumer should be
+                        # missing.
+                        self._partitions -= self._partitions
```

Comment:
See my comment on the pr-discussion thread - if `_remove_partitions` were safe (in the "only if we own them" sense), we could call it here. If it turns out that's not possible, the current way is best of course. Maybe in that case the comment could be expanded to "all ephemeral nodes for this should be missing, therefore we must not call _remove_partitions".